lsp/nls/src/linearization/mod.rs

```diff
@@ -139,8 +139,40 @@ impl Linearizer for AnalysisHost {
 
         let id = id_gen.get();
         match term {
+            Term::FunPattern(ident, destruct, _) => {
+                if let Some(ident) = ident {
+                    self.env.insert(ident.to_owned(), id);
+                    lin.push(LinearizationItem {
+                        id: id_gen.get_and_advance(),
+                        ty,
+                        pos: ident.pos.unwrap(),
+                        scope: self.scope.clone(),
+                        kind: TermKind::Declaration(ident.to_owned(), Vec::new()),
+                        meta: self.meta.take(),
```

Comment:
Do we want to use the metavalue of the function for the parameter? I think this is causing the last strange issue I have: when I hover on the first parameter of a function, it shows the documentation of the whole function as metadata.